diff --git a/app/Movies/Actor.php b/app/Movies/Actor.php
index 0ab4110..31711a5 100644
--- a/app/Movies/Actor.php
+++ b/app/Movies/Actor.php
@@ -9,8 +9,8 @@
          ->where('ACTOR_1',$term)
          ->orWhere('ACTOR_2',$term)
          ->orWhere('ACTOR_3',$term)
-           ->orderBy('YEAR','desc')
-           ->get();
+         ->orderBy('YEAR','desc')
+         ->paginate(12);
              return array('data' => $actors);
            }
          }
diff --git a/resources/views/layouts/actor.blade.php b/resources/views/layouts/actor.blade.php
index 52db195..bb486b6 100644
--- a/resources/views/layouts/actor.blade.php
+++ b/resources/views/layouts/actor.blade.php
@@ -18,5 +18,8 @@
 @endforeach
       </div>
     </div>
+    <div>
+      {{ $actors->links() }}
+    </div>
 
 @endsection
diff --git a/resources/views/layouts/app.blade.php b/resources/views/layouts/app.blade.php
index ce7c294..3d6d698 100644
--- a/resources/views/layouts/app.blade.php
+++ b/resources/views/layouts/app.blade.php
@@ -12,6 +12,7 @@
 
     <!-- Styles -->
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.0.13/css/all.css" integrity="sha384-DNOHZ68U8hZfKXOrtjWvjxusGo9WQnrNx2sqG0tfsghAvtVlRW3tvkXWZh58N9jp" crossorigin="anonymous">
 
 <!-- jQuery library -->
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
@@ -72,7 +73,11 @@
 .leave-rating input[type="radio"]:checked ~ label {
     color: #ffc600;
 }
-    </style>
+
+.mvs {
+  margin-top: 5px;
+}
+</style>
 </head>
 <body>
 
@@ -186,8 +191,9 @@
         </nav>
 
 
-
-        @yield('content')
+        <div class="container">
+          @yield('content')
+        </div>
 
 </body>
 </html>
diff --git a/resources/views/layouts/director.blade.php b/resources/views/layouts/director.blade.php
index b93dcc1..d274ba9 100644
--- a/resources/views/layouts/director.blade.php
+++ b/resources/views/layouts/director.blade.php
@@ -12,12 +12,15 @@
           @endif
         <p>{{ $direct->TITLE }}</p>
           <p>{{ $direct->YEAR }}</p>
-        
+
 
         </div>
         </div>
 @endforeach
       </div>
     </div>
+    <div>
+      {{ $director->links() }}
+    </div>
 
 @endsection
diff --git a/resources/views/layouts/movie.blade.php b/resources/views/layouts/movie.blade.php
index cfcccd9..b7d9a35 100644
--- a/resources/views/layouts/movie.blade.php
+++ b/resources/views/layouts/movie.blade.php
@@ -1,9 +1,9 @@
 <div class="movie">
   <div class="col-md-3">
   @if($movie->IMAGEPATH == "")
-    <a href="/movie/{{ $movie->ID }}">  <img src="/storage/posters/1.jpg" style="width:170px;height:250px" alt="default movie image" /> </a>
+    <a href="/movie/{{ $movie->ID }}">  <img class="img-thumbnail" src="/storage/posters/1.jpg" style="width:170px;height:250px" alt="default movie image" /> </a>
   @else
-    <a href="/movie/{{ $movie->ID }}">  <img class="img-responsive" style="width:170px;height:250px" src="/storage/posters/{{ $movie->IMAGEPATH }}" /> </a>
+    <a href="/movie/{{ $movie->ID }}">  <img class="img-thumbnail" style="width:170px;height:250px" src="/storage/posters/{{ $movie->IMAGEPATH }}" /> </a>
   @endif
 <p>{{ $movie->TITLE }}</p>
   <p>{{ $movie->YEAR }}</p>
diff --git a/resources/views/layouts/search.blade.php b/resources/views/layouts/search.blade.php
index 4b24eda..74e5720 100644
--- a/resources/views/layouts/search.blade.php
+++ b/resources/views/layouts/search.blade.php
@@ -2,77 +2,72 @@
 @section('content')
 </div>
 <div class="row">
-<div class="main_search" style="position:absolute;top:100px;right:550px">
-<div class="drop_menu_block">
-  <div class="col-md-3">
-<div class="drop_menu" style="position:absolute;right:50px"  >
-<p><b>Genre:</b></p>
-<select id="id1">
-<option value="|" selected="selected">All</option>
-<option value="action">Action</option>
-<option value="adventure">Adventure</option>
-<option value="animation">Animation</option>
-<option value="biography">Biography</option>
-<option value="comedy">Comedy</option>
-<option value="crime">Crime</option>
-<option value="documentary">Documentary</option>
-<option value="drama">Drama</option>
-<option value="family">Family</option>
-<option value="fantasy">Fantasy</option>
-<option value="film-noir">Film-Noir</option>
-<option value="game-show">Game-Show</option>
-<option value="history">History</option>
-<option value="horror">Horror</option>
-<option value="music">Music</option>
-<option value="musical">Musical</option>
-<option value="mystery">Mystery</option>
-<option value="news">News</option>
-<option value="reality-tv">Reality-TV</option>
-<option value="romance">Romance</option>
-<option value="sci-fi">Sci-Fi</option>
-<option value="sport">Sport</option>
-<option value="talk-show">Talk-Show</option>
-<option value="thriller">Thriller</option>
-<option value="war">War</option>
-<option value="western">Western</option>
-</select>
+  <div class="col-lg-4 col-md-4 col-sm-4 col-xs-4"></div>
+  <div class="col-lg-1 col-md-1 col-sm-1 col-xs-1">
+    <p><b>Genre:</b></p>
+    <select class="form-control" id="id1">
+      <option value="|" selected="selected">All</option>
+      <option value="action">Action</option>
+      <option value="adventure">Adventure</option>
+      <option value="animation">Animation</option>
+      <option value="biography">Biography</option>
+      <option value="comedy">Comedy</option>
+      <option value="crime">Crime</option>
+      <option value="documentary">Documentary</option>
+      <option value="drama">Drama</option>
+      <option value="family">Family</option>
+      <option value="fantasy">Fantasy</option>
+      <option value="film-noir">Film-Noir</option>
+      <option value="game-show">Game-Show</option>
+      <option value="history">History</option>
+      <option value="horror">Horror</option>
+      <option value="music">Music</option>
+      <option value="musical">Musical</option>
+      <option value="mystery">Mystery</option>
+      <option value="news">News</option>
+      <option value="reality-tv">Reality-TV</option>
+      <option value="romance">Romance</option>
+      <option value="sci-fi">Sci-Fi</option>
+      <option value="sport">Sport</option>
+      <option value="talk-show">Talk-Show</option>
+      <option value="thriller">Thriller</option>
+      <option value="war">War</option>
+      <option value="western">Western</option>
+    </select>
+  </div>
+  <div class="col-lg-1 col-md-1 col-sm-1 col-xs-1">
+    <p><b>Rating:</b></p>
+    <select class="form-control" id="id2">
+      <option value="-1" selected="selected">All</option>
+      <option value="9">9+</option>
+      <option value="8">8+</option>
+      <option value="7">7+</option>
+      <option value="6">6+</option>
+      <option value="5">5+</option>
+      <option value="4">4+</option>
+      <option value="3">3+</option>
+      <option value="2">2+</option>
+      <option value="1">1+</option>
+    </select>
+  </div>
+  <div class="col-lg-1 col-md-1 col-sm-1 col-xs-1">
+    <p><b>Order By:</b></p>
+    <select class="form-control"id="id3" >
+      <option value="YEAR" selected="selected">Year</option>
+      <option value="Oldest">Oldest</option>
+      <option value="Latest">Latest</option>
+      <option value="Alphabetical">Alphabetical</option>
+    </select>
+  </div>
+  <div class="col-lg-1 col-md-1 col-sm-1 col-xs-1">
+    <p>&nbsp</p>
+    <button id="search" class="btn btn-primary">Search</button>
+  </div>
+  <div class="col-lg-4 col-md-4 col-sm-4 col-xs-4"></div>
 </div>
+<div class="mvs">
 </div>
-<div class="col-md-3">
-<div class="drop_menu" style="position:absolute;right:-75px"  >
- <p><b>Rating:</b></p>
-<select id="id2">
-<option value="-1" selected="selected">All</option>
-<option value="9">9+</option>
-<option value="8">8+</option>
-<option value="7">7+</option>
-<option value="6">6+</option>
-<option value="5">5+</option>
-<option value="4">4+</option>
-<option value="3">3+</option>
-<option value="2">2+</option>
-<option value="1">1+</option>
-</select>
-</div>
-</div>
- <div class="col-md-3">
-<div class="drop_menu drop_menu-last" style="position:absolute;right:-100px" >
-<p><b>Order By:</b></p>
-<select class="selects-container"id="id3" >
-<option value="YEAR" selected="selected">Year</option>
-<option value="Oldest">Oldest</option>
-<option value="Latest">Latest</option>
-<option value="Alphabetical">Alphabetical</option>
-</select>
-</div>
-</div>
-</div>
-</div>
- <!--<input type="text" class="key"> -->
- <div style="position:absolute;right:350px;top:100px">
-      <button class="mybtn">Search</button>
-    </div>
-<div class="mvs" style="position:absolute;top:250px">
+<div id="nav-links">
 </div>
 <div id="nav-links">
 </div>
@@ -90,7 +85,6 @@ $(document).ready(function(){
       },
       success: function(result){
         $('.mvs').html("");
-        console.log(result);
         result['data']['data'].forEach(function(elem) {
           if (elem['IMAGEPATH']==""){
             elem['IMAGEPATH']='marina.jpg';
@@ -106,59 +100,82 @@ $(document).ready(function(){
         });
         next_page_url = result['data']['next_page_url'];
         previous_page_url = result['data']['prev_page_url'];
-        console.log(previous_page_url);
-        console.log(next_page_url);
         $("#nav-links").html("");
+
         if(previous_page_url){
-          var x = "<button class='btn btn-primary' id='left-btn' >Previous</button>"
+          var x = "<button style='margin-right: 5px;' class='btn btn-primary' id='left-btn'>Previous</button>"
           $("#nav-links").append(x);
           $("#left-btn").click(function() {
             get_movies(result['data']['current_page'] - 1);
           });
         }
+        var x = result["data"]['current_page'] + "/" + result["data"]["last_page"];
+        $("#nav-links").append(x);
         if(next_page_url) {
-          var x = "<button class='btn btn-primary' id='right-btn'>Next</button>"
+          var x = "<button style='margin-left: 5px;' class='btn btn-primary' id='right-btn'>Next</button>"
           $("#nav-links").append(x);
           $("#right-btn").click(function() {
             get_movies(result['data']['current_page'] + 1);
           });
         }
     }
-  });
-}
+    });
+  }
   get_movies(1);
+  function get_movies_filter(page) {
+    $.ajax({
+      type: "GET",
+      url: '/api/orderby2?t=' + $('#id3').val() + '&r=' + $('#id2').val() + '&g=' + $('#id1').val() + "&page=" + page,
+      data:{},
+      dataType: "json",
+      error: function (request, error) {
+        alert(error);
+        alert(request);
+      },
+      success: function(result){
+        $('.mvs').html("");
+        $("#nav-links").html("");
+        console.log(result);
 
-$('.mybtn').on('click',function(){
-$.ajax({
-  type: "GET",
-    url: '/api/orderby2?t=' + $('#id3').val() + '&r=' + $('#id2').val() + '&g=' + $('#id1').val(),
-  data:{},
-    dataType: "json",
-    error: function (request, error) {
-alert(error);
-alert(request);
-  },
-    success: function(result){
-      $('.mvs').html("");
-
-      result['data']['data'].forEach(function(elem) {
-        if (elem['IMAGEPATH']=="")
-            { elem['IMAGEPATH']='marina.jpg'; }
+        result['data']['data'].forEach(function(elem) {
+          if (elem['IMAGEPATH']=="")
+              { elem['IMAGEPATH']='marina.jpg'; }
 
-      var x = `
-      <div class="col-md-3" style="height:350px" >
-      <a href="/movie/`+elem['ID']+`"><img class="img-thumbnail" src="/storage/posters/`+elem['IMAGEPATH']+`"  alt="marina" width="170" height="255"> </a>
-        <p style="height:20px"> <strong>Title : </strong> ` + elem['TITLE'] + `</p>
-        <p style="height:50px"><strong>Year : </strong> ` + elem['YEAR'] + `</p>
-        </div>
-          `;
+          var x = `
+            <div class="col-md-3" style="height:350px" >
+            <a href="/movie/`+elem['ID']+`"><img class="img-thumbnail" src="/storage/posters/`+elem['IMAGEPATH']+`"  alt="marina" width="170" height="255"> </a>
+            <p style="height:20px"> <strong>Title : </strong> ` + elem['TITLE'] + `</p>
+            <p style="height:50px"><strong>Year : </strong> ` + elem['YEAR'] + `</p>
+            </div>`;
 
-$('.mvs').append(x);
+          $('.mvs').append(x);
+          next_page_url = result['data']['next_page_url'];
+          previous_page_url = result['data']['prev_page_url'];
+          $("#nav-links").html("");
 
-});
-}
-});
-});
+          if(previous_page_url){
+            var x = "<button style='margin-right: 5px;' class='btn btn-primary' id='left-btn'>Previous</button>"
+            $("#nav-links").append(x);
+            $("#left-btn").click(function() {
+              get_movies_filter(result['data']['current_page'] - 1);
+            });
+          }
+          var x = result["data"]['current_page'] + "/" + result["data"]["last_page"];
+          $("#nav-links").append(x);
+          if(next_page_url) {
+            var x = "<button style='margin-left: 5px;' class='btn btn-primary' id='right-btn'>Next</button>"
+            $("#nav-links").append(x);
+            $("#right-btn").click(function() {
+              get_movies_filter(result['data']['current_page'] + 1);
+            });
+          }
+        });
+      }
+      });
+  }
+  $('#search').click(function() {
+    get_movies_filter(1);
+  });
 });
 
 </script>
diff --git a/routes/api.php b/routes/api.php
index 2f53e45..d7495c4 100644
--- a/routes/api.php
+++ b/routes/api.php
@@ -198,7 +198,7 @@ if ($query == 'Latest'){
  $order = DB::table('movies')->select ('TITLE', 'YEAR','IMAGEPATH','ID')
                ->where('GENERS','LIKE',"%$query2%")
                ->orderBy('created_at','asc')
-               ->get();
+               ->paginate(12);
 }
 }
   return response()->json (['status'=>'success',
